static/js/components/widgets/ResourcePickerDialog.tsx

```diff
@@ -151,13 +107,30 @@ const modeText = {
 
 export default function ResourcePickerDialog(props: Props): JSX.Element {
   const { mode, isOpen, closeDialog, insertEmbed, contentNames } = props
+  const website = useWebsite()
+  const definedCategories = useMemo(() => {
+    return website.starter?.config?.collections
+      .filter(entry => entry.category === "Content")
+      .flatMap(elt => [elt.name, elt.label])
+  }, [website.starter?.config?.collections])
```

Comment:
This has structure like
```
["pages", "Pages", "bananas", "Bananas", "video_galleries", "Video Galleries", ...]
```
so implicitly, indices 0, 1 are a single item, 2, 3 are a single item, etc.

**Requested change:** Make that explicit: group the things together.

**Suggestion:** I think a structure like
```ts
const contentCategires = {
  pages: { name: page, label: "Page" }, 
  bananas: { name: bananas, label: "Bananas" },
  ...
}
```
would be helpful (more explicit, and also easier lookups below... you won't have to fuss with `indexOf`.)

lodash's `[keyBy](https://lodash.com/docs/4.17.15#keyBy)` is very helpful for this sort of thing...something like
```ts
  const definedCategories = useMemo(() => {
    const contentCollections = website.starter?.config?.collections
      .filter(entry => entry.category === "Content") ?? []
    return keyBy(contentCollections, collection => collection.name)
  }, [website.starter?.config?.collections])
```

Reply:
Updated in https://github.com/mitodl/ocw-studio/pull/1448/commits/b46ad80309a35e46f7487f9c547179a8bcbe6fe2.